### ISSUE TITLE
 AFFLIB 3.7.17: FAIL: aftest

### ISSUE THREAD

**Issue by pacho2** [sshock/AFFLIBv3#37]
I get a test failing while building:
FAIL: aftest
============================================================================
Testsuite summary for AFFLIB 3.7.17
============================================================================
# TOTAL: 1
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
============================================================================
See lib/test-suite.log
Please report to https://github.com/sshock/AFFLIBv3
============================================================================
make[3]: *** [Makefile:1314: test-suite.log] Error 1
make[3]: Leaving directory '/var/tmp/portage/app-forensics/afflib-3.7.17/work/AFFLIBv3-3.7.17/lib'
make[2]: *** [Makefile:1422: check-TESTS] Error 2
make[2]: Leaving directory '/var/tmp/portage/app-forensics/afflib-3.7.17/work/AFFLIBv3-3.7.17/lib'
make[1]: *** [Makefile:1495: check-am] Error 2
make[1]: Leaving directory '/var/tmp/portage/app-forensics/afflib-3.7.17/work/AFFLIBv3-3.7.17/lib'
make: *** [Makefile:566: check-recursive] Error 1
[test-suite.log](https://github.com/sshock/AFFLIBv3/files/2799914/test-suite.log)

**Comment by sshock** [comment]
The tests broke as a side effect of fix #29 made in Sep 2017 (see issue #28 for more details).

The quick summary is that the first 1000 segments are easy (.000 to .999), but after that it goes to .A00 and there was a discrepancy between what the code said it would do and what it actually did for incrementing after that, so we fixed the code to match what it said it would do, which is also what makes the most sense.

The sad thing is, although I have noticed the `aftest.cpp` file before, I never actually looked at it or built it, so I didn't even know it was unit tests or else maybe I would have thought about running it once in a while ;)

Anyway, I'll fix this soon (probably later tonight); it will be real easy to tweak what extensions it is checking for in aftest.cpp.
